cmd/embedded-cluster/restore.go

```diff
@@ -886,370 +882,364 @@ func installAndWaitForRestoredK0sNode(c *cli.Context, applier *addons.Applier) (
-// installation from the flag.
-func restoreReconcileLocalArtifactMirrorPortFromFlag(c *cli.Context) error {
+// restoreReconcileInstallationFromRuntimeConfig will update the installation to match the runtime
+// config from the original installation.
+func restoreReconcileInstallationFromRuntimeConfig(ctx context.Context, runtimeConfig *ecv1beta1.RuntimeConfigSpec) error {
 	kcli, err := kubeutils.KubeClient()
 	if err != nil {
 		return fmt.Errorf("create kube client: %w", err)
 	}
-	in, err := kubeutils.GetLatestInstallation(c.Context, kcli)
+	in, err := kubeutils.GetLatestInstallation(ctx, kcli)
 	if err != nil {
 		return fmt.Errorf("get latest installation: %w", err)
 	}
-	if in.Spec.LocalArtifactMirror == nil {
-		in.Spec.LocalArtifactMirror = &ecv1beta1.LocalArtifactMirrorSpec{}
-	}
-	port, err := getLocalArtifactMirrorPortFromFlag(c)
-	if err != nil {
-		return fmt.Errorf("get local artifact mirror port: %w", err)
-	}
-	if in.Spec.LocalArtifactMirror.Port == port {
-		return nil
-	}
-	logrus.Debugf("updating local artifact mirror port from flag to %d on installation %q", port, in.Name)
-	in.Spec.LocalArtifactMirror.Port = port
-	if err := kcli.Update(c.Context, in); err != nil {
+
+	// We do not support changing these values so we must choose the values from the installation
+	// prior to the restore.
+	in.Spec.RuntimeConfig.DataDir = runtimeConfig.DataDir
+	in.Spec.RuntimeConfig.K0sDataDirOverride = runtimeConfig.K0sDataDirOverride
+
+	// We allow the user to override the port with a flag to the restore command.
+	in.Spec.RuntimeConfig.LocalArtifactMirror.Port = runtimeConfig.LocalArtifactMirror.Port
+
+	if err := kcli.Update(ctx, in); err != nil {
 		return fmt.Errorf("update installation: %w", err)
 	}
 	return nil
 }
 
-// restoreReconcileLocalArtifactMirrorPortFromBackup will update the service to use the port from
-// the installation.
-func restoreReconcileLocalArtifactMirrorPortFromBackup(backup *velerov1.Backup) error {
-	portStr := backup.Annotations["kots.io/embedded-cluster-local-artifact-mirror-port"]
-	if portStr == "" {
-		return nil
-	}
-	port, err := k8snet.ParsePort(portStr, false)
-	if err != nil {
-		return fmt.Errorf("unable to parse local artifact mirror port from backup: %w", err)
-	}
-	logrus.Debugf("updating local artifact mirror port from backup to %d", port)
-	if err := updateLocalArtifactMirrorService(port); err != nil {
-		return fmt.Errorf("unable to update local artifact mirror service: %w", err)
+// overrideRuntimeConfigFromBackup will update the runtime config from the backup. These values may
+// be used during the install and set in the Installation object via the
+// restoreReconcileInstallationFromRuntimeConfig function.
+func overrideRuntimeConfigFromBackup(c *cli.Context, runtimeConfig *ecv1beta1.RuntimeConfigSpec, backup *velerov1.Backup) (*ecv1beta1.RuntimeConfigSpec, error) {
+	if !c.IsSet("local-artifact-mirror-port") {
```

Comment:
i don't see where this flag is used / updates the runtime config

Reply:
It is set in the urfave/cli flag action

https://github.com/replicatedhq/embedded-cluster/pull/1238/files#diff-114d16ace9fc8606c3b1e9a3ba899010aae12b04b9ab398f4973a52e1e63f55dR55